lib/hanami/events/handler_runner.rb

```diff
@@ -9,7 +9,9 @@ def initialize(handler, logger)
       end
 
       def call(payload)
-        instance_exec(payload, &@handler)
+        return instance_exec(payload, &@handler) if @handler.is_a?(Proc)
+
+        @handler.call(payload)
```

Comment:
and with this case, we can drop unnecessary code and `events/base.rb` file will be simpler